sunpy/visualization/imageanimator.py

```diff
@@ -608,32 +585,257 @@ def _sanitize_axis_range(self, axis_range, data):
                 # If min==max or None
                 if axis_range[i] is None or axis_range[i][0] == axis_range[i][1]:
                     if i in self.slider_axes:
-                        axis_range[i] = np.linspace(0,d,d)
+                        axis_range[i] = np.linspace(0, d, d)
                     else:
                         axis_range[i] = [0, d]
                         # min max pair for slider axes should be converted
                         # to an array
                 elif i in self.slider_axes:
                     axis_range[i] = np.linspace(axis_range[i][0], axis_range[i][1], d)
 
-            #If we have a whole list of values for the axis, make sure we are a slider axis.
+            # If we have a whole list of values for the axis, make sure we are a slider axis.
             elif len(axis_range[i]) == d:
-                if i not in self.slider_axes:
-                    raise ValueError("Slider axes mis-match, non-slider axes need [min,max] pairs")
-                else:
-                    #Make sure the resulting element is a ndarray
-                    axis_range[i] = np.array(axis_range[i])
+                axis_range[i] = np.array(axis_range[i])
 
-            #panic
+            # panic
             else:
-                raise ValueError("axis_range should be either: None, [min,max], or a linspace for slider axes")
+                raise ValueError("axis_range must be None or a list with length equal to number "
+                                 "of axes in data whose elements are either None, [min,max], "
+                                 "or a list/array of same length as the plot/image axis of data.")
         return axis_range
 
-    def _updateimage(self, val, im, slider):
+
+class ImageAnimator(ArrayAnimator):
+    """Create a matplotlib backend independent data explorer for 2D images.
+
+    The following keyboard shortcuts are defined in the viewer:
+
+    - 'left': previous step on active slider
+    - 'right': next step on active slider
+    - 'top': change the active slider up one
+    - 'bottom': change the active slider down one
+    - 'p': play/pause active slider
+
+    This viewer can have user defined buttons added by specifying the labels
+    and functions called when those buttons are clicked as keyword arguments.
+
+    Parameters
+    ----------
+    data: ndarray
+        The data to be visualised >= 2D
```

Comment:
You mean you challenge that SunPy developer instructions that say to use Americanized spelling??  (I learned something while copying that stuff across to irispy ;) )